### PR TITLE
feat: add to_kubernetes_yaml to Deployment

### PR DESCRIPTION
**Pull request by alaeddine-13**
closes: https://github.com/jina-ai/jina/issues/5584

### CODE REVIEW

File: jina/orchestrate/deployments/__init__.py [hunk 1]
```diff
@@ -134,14 +134,13 @@ def __init__(
         metrics_exporter_host: Optional[str] = None, 
         metrics_exporter_port: Optional[int] = None, 
         monitoring: Optional[bool] = False, 
-        name: Optional[str] = None, 
+        name: Optional[str] = 'executor',
```
Comment (JoanFM) [inline jina/orchestrate/deployments/__init__.py:137]: weird

**Review by JoanFM — CHANGES_REQUESTED** [review]
Check the comment, and the prefer-platform issue

**Comment by alaeddine-13** [comment]
> Check the comment, and the prefer-platform issue

I don't think there is anything wrong with it. This argument shouldn't exist because prefer_platform does not exist for executors. Try: jina executor --help

**Review by JoanFM — APPROVED** [review]

File: docs/concepts/executor/serve.md [hunk 2]
```diff
@@ -157,6 +155,33 @@ The above example deploys the `DummyHubExecutor` from Executor Hub into your Kub
 The Executor you use needs to be already containerized and stored in a registry accessible from your Kubernetes cluster. We recommend Executor Hub for this.
 ````
 
+Once the Executor is deployed, you can expose a service:
```
Comment (girishc13) [inline docs/concepts/executor/serve.md:158]: What's the reason behind not exposing the service automatically?

Comment (girishc13) [reply docs/concepts/executor/serve.md:158]: Nevermind. This is only for the external IP.

File: docs/concepts/executor/serve.md [hunk 3]
```diff
@@ -157,6 +155,33 @@ The above example deploys the `DummyHubExecutor` from Executor Hub into your Kub
...
+```bash
+kubectl expose deployment executor --name=executor-exposed --type LoadBalancer --port 80 --target-port 8080 -n my-namespace
+sleep 60 # wait until the external ip is configured
+```
+
+Let's export the external IP address created and use it to send requests to the Executor. 
+```bash
+export EXTERNAL_IP=`kubectl get service executor-exposed -n my-namespace -o=jsonpath='{.status.loadBalancer.ingress[0].ip}'`
+```
+To send requests using {meth}`~jina.Client`, make sure to set `stream=False`:
```
Comment (girishc13) [inline docs/concepts/executor/serve.md:168]: If a Gateway is added by default then there is no need to set `stream=False`.

File: docs/concepts/executor/serve.md (hunk 2 above)
Comment (alaeddine-13) [reply docs/concepts/executor/serve.md:158]: I guess it depends on how the user wants to do so, which varies from kubernetes provider to another.
 I guess the service we expose already is mostly useful for jcloud. Not sure how a user would leverage it without portforwarding

File: docs/concepts/executor/serve.md (hunk 3 above)
Comment (girishc13) [reply docs/concepts/executor/serve.md:168]: My misunderstanding again. The `include_gateway` is used only in the overload method in the Flow.

**Review by girishc13 — DISMISSED** [review]

**Review by JoanFM — DISMISSED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**